### ISSUE TITLE
Can't close stable position due to not enough inbound capacity

### ISSUE THREAD

**Issue by holzeis** [get10101/10101#1257]
I have opened a stable position and am unable to close it again due to 

app.logs
```
Could not handle trade request: Could not propose dlc channel: Invalid parameters were provided: Not enough inbound capacity to establish given contract.
```

It appears the coordinator thinks I want to open a new position, although I am already having a position. Luckily I do not have enough capacity to execute this trade. I guess this issue is related to me having two channels one collaboratively closed and the other active.

coordinator.logs
```
{"timestamp":"2023-09-08T11:25:29.063722622Z","level":"INFO","fields":{"message":"Opening position","peer_id":"02aa6fc81e341d69b36174365644a3560bfcc51ce0fb9bb67d93712bc54117cbb8","trade_params":"TradeParams { pubkey: PublicKey(b8cb1741c52b71937db69bfbe01cc5fc0b56a34456367461b3691d341ec86faaeae307d7d40d5cfa9e62f710787e54973c2b5b0e54f1f36a9b7e64330a223d4e), contract_symbol: BtcUsd, leverage: 1.0, quantity: 10.0, direction: Long, filled_with: FilledWith { order_id: b8e19a86-1bf0-46eb-af04-206f8626e4d2, expiry_timestamp: 2023-09-15 0:00:00.0 +00:00:00, oracle_pk: XOnlyPublicKey(c071e1fd881a55831cc38c85986c6c72193b4e3a98bcbc460f6c1ed2f78cf8167458a18010faf718756c69de33ee9e38a0d3f0ad48befa58e23a4b1f843e08be), matches: [Match { order_id: 68e56480-d0bf-4a87-b4b4-081010abe45e, quantity: 10, pubkey: PublicKey(8642f644c8c754c1c3266a92c8bc31514f7e3207aa3b1e084f8cba48e67defd9f985afda441e954c8885586a5b4e9f15f25a63008737989b131a9f30317e5f63), execution_price: 25840.5 }] } }"},"target":"coordinator::node","span":{"name":"post_trade"},"spans":[{"name":"post_trade"}]}
```

**Comment by holzeis** [comment]
My DLC channel seems to be `OffChainClosed` on the coordinator, although its open on my phone.

```
$ curl localhost:8000/api/admin/dlc_channels | jq '.[] | select(.counter_party == "02aa6fc81e341d69b36174365644a3560bfcc51ce0fb9bb67d93712bc54117cbb8")'
  % Total    % Received % Xferd  Average Speed   Time    Time     Time  Current
                                 Dload  Upload   Total   Spent    Left  Speed
100 14188  100 14188    0     0  2857k      0 --:--:-- --:--:-- --:--:-- 3463k
{
  "channel_id": "4ac4b7ff5b79c6cd4f06ad2f4e718dcd0364bb729d5e47f81dcf92f1cdbc3055",
  "dlc_channel_id": null,
  "counter_party": "02aa6fc81e341d69b36174365644a3560bfcc51ce0fb9bb67d93712bc54117cbb8",
  "update_idx": 281474976710655,
  "state": "OffChainClosed",
  "fee_rate_per_vb": 10,
  "fund_value_satoshis": 100000,
  "is_offer": true,
  "user_email": "richard@10101.finance"
}
```

**Comment by holzeis** [comment]
It seems my position got closed due to expiry. I guess I opened the stable position before we have upgraded the coordinator. When closing the position we failed after sending the `SubChannelCloseFinalize ` setting the coordinator to done.


The protocol did not finish successfully on both sides. The coordinator got until the last message `SubChannelCloseFinalize`, but did not send the `SubChannelRevoke` message. I guess this message got lost due to the disconnect right after processing the `SubChannelCloseFinalize. The coordinator should actually resend the `SubChannelRevoke` message but did not do so, thus leaving my app in a limbo state.

coordinator.logs
```
{"timestamp":"2023-09-01T11:39:03.261653041Z","level":"DEBUG","fields":{"message":"Marking channels with 02aa6fc81e341d69b36174365644a3560bfcc51ce0fb9bb67d93712bc54117cbb8 disconnected and generating channel_updates.","log.target":"lightning::ln::channelmanager","log.module_path":"lightning::ln::channelmanager","log.file":"/home/runner/.cargo/git/checkouts/rust-lightning-bee3233422fa4387/3b69cec/lightning/src/ln/channelmanager.rs","log.line":6493},"target":"lightning::ln::channelmanager"}
{"timestamp":"2023-09-01T11:38:32.443032966Z","level":"INFO","fields":{"message":"Processing message","from":"02aa6fc81e341d69b36174365644a3560bfcc51ce0fb9bb67d93712bc54117cbb8","kind":"SubChannelCloseFinalize"},"target":"coordinator::node"}
{"timestamp":"2023-09-01T11:38:32.040703358Z","level":"INFO","fields":{"message":"Sending message","to":"02aa6fc81e341d69b36174365644a3560bfcc51ce0fb9bb67d93712bc54117cbb8","kind":"SubChannelCloseConfirm"},"target":"coordinator::node"}
{"timestamp":"2023-09-01T11:38:32.037093782Z","level":"INFO","fields":{"message":"Processing message","from":"02aa6fc81e341d69b36174365644a3560bfcc51ce0fb9bb67d93712bc54117cbb8","kind":"SubChannelCloseAccept"},"target":"coordinator::node"}
{"timestamp":"2023-09-01T11:38:32.037093782Z","level":"INFO","fields":{"message":"Processing message","from":"02aa6fc81e341d69b36174365644a3560bfcc51ce0fb9bb67d93712bc54117cbb8","kind":"SubChannelCloseAccept"},"target":"coordinator::node"}
{"timestamp":"2023-09-01T11:38:31.59842684Z","level":"INFO","fields":{"message":"Successfully proposed to close expired position with 02aa6fc81e341d69b36174365644a3560bfcc51ce0fb9bb67d93712bc54117cbb8"},"target":"coordinator::node::expired_positions"}
{"timestamp":"2023-09-01T11:38:31.591913963Z","level":"DEBUG","fields":{"message":"Closing position of 3808 with 02aa6fc81e341d69b36174365644a3560bfcc51ce0fb9bb67d93712bc54117cbb8","position":"Position { id: 50, contract_symbol: BtcUsd, leverage: 2.0, quantity: 2.0, direction: Short, average_entry_price: 26404.0, liquidation_price: 52808.0, position_state: Open, collateral: 7575, creation_timestamp: 2023-06-17 17:58:30.352947 +00:00:00, expiry_timestamp: 2023-06-19 0:00:00.0 +00:00:00, update_timestamp: 2023-09-01 11:33:36.502987 +00:00:00, trader: PublicKey(b8cb1741c52b71937db69bfbe01cc5fc0b56a34456367461b3691d341ec86faaeae307d7d40d5cfa9e62f710787e54973c2b5b0e54f1f36a9b7e64330a223d4e), temporary_contract_id: None, closing_price: None }","channel_id":"4ac4b7ff5b79c6cd4f06ad2f4e718dcd0364bb729d5e47f81dcf92f1cdbc3055","accept_settlement_amount":"3808"},"target":"coordinator::node"}
```


app.logs
```
2023-09-01 13:38:32.264  LogLevel.INFO  Sending message kind: SubChannelCloseFinalize,to: 022ae8dbec1caa4dac93f07f2ebf5ad7a5dd08d375b79f11095e81b065c2155156  
2023-09-01 13:38:32.265  LogLevel.WARNING  Could not find a filling position in the database. Maybe because the coordinator closed an expired position. Error: There is no order in state filling in the database   
2023-09-01 13:38:32.265  LogLevel.DEBUG  Adding order for the expired closed position   
2023-09-01 13:38:32.266  LogLevel.ERROR  Critical Error! We have a DLC but were unable to set the order to filled. Error: timestamp must not be in the future!   
```

**Comment by holzeis** [comment]
Although this is mainly a protocol issue, the error only happened because I wasn't aware that my position got closed due to expiry. #1246 will improve that and should mitigate the risk of that bug from happening.

**Comment by holzeis** [comment]
It looks like the subchannel was also successfully closed on the app side. Printing the subchannels show that I have two subchannels

1. OffChainClose - what it should be.
2. CounterOnChainClosed - from the collaboratively closed channel before (old).

```
flutter: 2023-09-10 11:20:42.685  LogLevel.DEBUG  --------> SubChannel { channel_id: [74, 196, 183, 255, 91, 121, 198, 205, 79, 6, 173, 47, 78, 113, 141, 205, 3, 100, 187, 114, 157, 94, 71, 248, 29, 207, 146, 241, 205, 188, 48, 85], state: OffChainClosed }
flutter: 2023-09-10 11:20:42.685  LogLevel.DEBUG  --------> SubChannel { channel_id: [182, 75, 149, 192, 108, 65, 229, 107, 232, 90, 37, 222, 61, 111, 180, 22, 194, 197, 188, 253, 188, 212, 115, 100, 1, 156, 160, 118, 150, 134, 16, 160], state: CounterOnChainClosed }
```

**Comment by holzeis** [comment]
After manually removing the outdated position. It turns out the calculated `settlement_amount` was far off.

The following was my actual position!
```
115 | BtcUsd          |        1 |       10 | short     |             26007.5 |      1.048575e+06 | Closed         |      38450 | 2023-09-01 11:37:43.100334+00 | 2023-09-08 00:00:00+00 | 2023-09-01 11:39:03.436335+00 | 02aa6fc81e341d69b36174365644a3560bfcc51ce0fb9bb67d93712bc54117cbb8 | 5b2693da0572eace3ca25b85c0903928af5d927023ff7a87dfeb0d13a6ee967b |            34642 |                    |         26332
```

But it seems this got closed. (see also that both have the exact same closing price)
```
50 | BtcUsd          |        2 |        2 | short     |               26404 |             52808 | Closing        |       7575 | 2023-06-17 17:58:30.352947+00 | 2023-06-19 00:00:00+00 | 2023-09-10 09:39:07.723899+00 | 02aa6fc81e341d69b36174365644a3560bfcc51ce0fb9bb67d93712bc54117cbb8 |                                                                  |                  |                -21 |         26332
```

coordinator logs
```
Closing position of 3808 with 02aa6fc81e341d69b36174365644a3560bfcc51ce0fb9bb67d93712bc54117cbb8","position":"Position { id: 50, contract_symbol: BtcUsd, leverage: 2.0, quantity: 2.0, direction: Short, average_entry_price: 26404.0, liquidation_price: 52808.0, position_state: Open, collateral: 7575, creation_timestamp: 2023-06-17 17:58:30.352947 +00:00:00, expiry_timestamp: 2023-06-19 0:00:00.0 +00:00:00, update_timestamp: 2023-09-01 11:33:36.502987 +00:00:00, trader: PublicKey(b8cb1741c52b71937db69bfbe01cc5fc0b56a34456367461b3691d341ec86faaeae307d7d40d5cfa9e62f710787e54973c2b5b0e54f1f36a9b7e64330a223d4e), temporary_contract_id: None, closing_price: None }","channel_id":"4ac4b7ff5b79c6cd4f06ad2f4e718dcd0364bb729d5e47f81dcf92f1cdbc3055","accept_settlement_amount":"3808"},"target":"coordinator::node"
```

I guess what happened was that I had (due to an erroneous state before) two open positions. (one from a broken state from a collab closed channel and the other one which was legit). Our `expired_positions` logic, however looks for any expired open position. 
- The old open position has been expired (but not the new one) resulting in an incorrect `settlement_amount` and a premature closing of the dlc channel.
- The app only new of the new position and rejected to delete the position as the expiry time was still in the future (however the dlc still got closed in the background leaving us with an inconsistent state).

**Comment by holzeis** [comment]
I don't think we need to directly action something on that ticket, as the bug happened due to an erronous state and code, which is already outdated. I guess adding the dlc_channel a position belongs would help, that way we could add sanity checks as well as set all positions to closed which do not refer to a signed `dlc_channel`.

**Comment by holzeis** [comment]
> At least we should check if other users have a similar state.

I found 4 other users with multiple open positions.

```
orderbook=# select trader_pubkey, position_state from positions where position_state='Open' group by trader_pubkey, position_state having count(*) > 1;
                           trader_pubkey                            | position_state 
--------------------------------------------------------------------+----------------
 021070ab75d6ef553bca3ba710c0d97c413725e97f469977cebf1d42bbca626aa6 | Open
 034733d968b1cd197e41fb4eda0c2c5f46a457428d1a80075d5e0790520ca88282 | Open
 02d12c7036199af86e2eeda7d396230f11f93d74400978e002d006f9d2290e3071 | Open
 03215c55a4d88b14cf659ffde5eb92e68eae6d5b83ad5310bd9742a8cf6564068f | Open
(4 rows)
```

2 of them have a `dlc_channel` which is stuck in the `Offered` state and the other 2 did not have a `dlc_channel` at all. I've set all those positions to Closed to be on the safe side.

**Comment by holzeis** [comment]
Closing this ticket with the following conclusion.

The error not enough inbound capacity was thrown correctly. The issue was, that I had 2 positions in the state `Open`. Out outdated expiry logic tried to close this outdated position on my new `dlc_channel`. The protocol succeeded, but removing the position on the app failed, since the expiry was not reached on the new position yet. The error `Not enough inbound capacity` was thrown because my app submitted a counter order for my (seemingly) still open position. The coordinator though I wanted to open a new position, but given that the `settlement_amount` on my outdated position was so low, I did not have enough liquidity to perform that trade.

I am closing this ticket as I've fixed the incorrect data.
